x/crosschain/keeper/processs_outbound.go

```diff
@@ -48,12 +59,71 @@ ProcessFailedOutbound processes a failed outbound transaction. It does the follo
 
  4. Set the finalization status of the current outbound tx to executed. If a revert tx is is created, the finalization status is not set, it would get set when the revert is processed via a subsequent transaction
 */
+
+// This function sets CCTX status , in cases where the outbound tx is successful, but tx itself fails
+// This is done because SaveSuccessfulOutbound does not set the cctx status
+// For cases where the outbound tx is unsuccessful, the cctx status is automatically set to Aborted in the ProcessFailedOutbound function, so we can just return and error to trigger that
 func (k Keeper) ProcessFailedOutbound(ctx sdk.Context, cctx *types.CrossChainTx, valueReceived string) error {
 	oldStatus := cctx.CctxStatus.Status
-	if cctx.InboundTxParams.CoinType == coin.CoinType_Cmd || chains.IsZetaChain(cctx.InboundTxParams.SenderChainId) {
+	if cctx.InboundTxParams.CoinType == coin.CoinType_Cmd {
 		// if the cctx is of coin type cmd or the sender chain is zeta chain, then we do not revert, the cctx is aborted
 		cctx.GetCurrentOutTxParam().TxFinalizationStatus = types.TxFinalizationStatus_Executed
-		cctx.CctxStatus.ChangeStatus(types.CctxStatus_Aborted, "")
+		cctx.SetAbort("Outbound failed")
+	} else if chains.IsZetaChain(cctx.InboundTxParams.SenderChainId) {
```

Comment:
I think this logic should be put in a separate function but can be addressed in a further PR

Reply:
Do you mean the handling for coin type CMD ?

Reply:
Everything under `else if chains.IsZetaChain(cctx.InboundTxParams.SenderChainId) {`

Reply:
aee29163c0e0ca8d8d2b2c7538e792764367339f